### PR TITLE
Format extension types.

### PR DESCRIPTION
**Pull request by munificent**
Nothing particularly interesting here, but it did flush out a little bug in `PieceFactory.createParameter()` where a comment before the metadata annotation would get moved to after the parameter.

### CODE REVIEW

**Review by natebosch — APPROVED** [review]

File: lib/src/piece/list.dart [hunk 1]
```diff
@@ -487,3 +487,10 @@ class ListStyle {
       this.splitListIfBeforeSplits = false,
       this.allowBlockElement = false});
 }
+
```
Comment (kallentu) [inline lib/src/piece/list.dart:490]: Heh, what's happening with this enum? Probably remove this too ahah

File: test/declaration/extension_type_comment.unit [hunk 2]
```diff
@@ -0,0 +1,61 @@
+40 columns                              |
+>>> Inline block comments everywhere.
```
Comment (kallentu) [inline test/declaration/extension_type_comment.unit:2]: Wow lol these tests are way more extensive than our typical comment tests.

**Review by kallentu — APPROVED** [review]

File: lib/src/piece/list.dart (hunk 1 above)
Comment (munificent) [reply lib/src/piece/list.dart:490]: Heh, oops! I was seeing how the current formatter behaved with some temp code. Removed.

File: test/declaration/extension_type_comment.unit (hunk 2 above)
Comment (munificent) [reply test/declaration/extension_type_comment.unit:2]: Yeah, Lasse wrote these and I just migrated them over.

**Pull request merged**